### ISSUE TITLE
[Bug] Building the tests should force building the client and server

### ISSUE THREAD

**Issue by 0x00002a** [Tectu/malloy#61]
Currently it is possible to tell cmake to build the tests without either the client, server, or (I'm guessing) both. This results in a linking error since the tests link to both the server and client. 

Unless we want to split up the tests (which might prove difficult and which I can see no benefits for, upstream packages don't run the tests on build I'd hope), this should not be an allowed configuration. I suggest having it fail rather than (silently or not) toggle the flags automatically.

**Comment by Tectu** [comment]
I'm unsure as to why this should be the case. The test suite executable `malloy-tests` links to both `malloy-client` and `malloy-server`:

https://github.com/Tectu/malloy/blob/a6dccf81ee24dec60433309f3fd85237eb120c7b/test/CMakeLists.txt#L23

On my side both the client & server get built (along with their `malloy-core` dependency) when doing a clean build on the `malloy-tests` target.

**Comment by 0x00002a** [comment]
Yes but this line: https://github.com/Tectu/malloy/blob/a6dccf81ee24dec60433309f3fd85237eb120c7b/lib/malloy/CMakeLists.txt#L5 removes the part that defines `malloy-client` as a target, meaning its just a library to cmake at that point and cmake allows libraries it can't identify in `target_link_libraries`

**Comment by Tectu** [comment]
I see what you mean.
Yeah, this definitely needs some improvements. Currently I think the best option is indeed to just have the `malloy-tests` cmake target fail if client & server are not both enabled. I'm not sure whether there will be undesired side effects but as you mentioned that shouldn't be a problem for anyone not actively working on `malloy`.
